WcaOnRails/app/webpacker/components/RolesTab/RoleForm.jsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+
+import { Form } from 'semantic-ui-react';
+
+import useLoadedData from '../../lib/hooks/useLoadedData';
+import { roleDataUrl, roleUpdateUrl } from '../../lib/requests/routes.js.erb';
+import useSaveAction from '../../lib/hooks/useSaveAction';
+import Errored from '../Requests/Errored';
+import DelegateForm from './DelegateForm';
+
+const groups = [{
+  key: 'delegates',
+  text: 'Delegates',
+  value: 'Delegates',
+}];
+
+const endRoleValue = {
+  delegateStatus: '',
+  seniorDelegateId: '',
+  location: '',
+};
+
+export default function RoleForm() {
+  const userId = window.location.pathname.split('/')[2];
+  const roleId = window.location.pathname.split('/')[4];
+  const { data, loading, error } = useLoadedData(roleDataUrl(userId, roleId));
+  const { save } = useSaveAction();
+  const [selectedGroup] = React.useState(groups[0].value);
+  const [formValues, setFormValues] = React.useState({});
+  const [apiError, setError] = React.useState(false);
+  const [saving, setSaving] = React.useState(false);
+
+  const updateRole = (endRole = false) => {
+    setSaving(true);
+    save(
+      roleUpdateUrl,
+      {
+        userId,
+        roleId,
+        ...(endRole ? endRoleValue : formValues),
```

Comment:
It's a bit confusing that `endRole` is a function name and the function param name.